src/applications/simple-forms-form-engine/shared/tests/unit/config/pages/customStepPage.unit.spec.js

```diff
@@ -67,24 +76,103 @@ describe('customStepPage', () => {
     const { required } = pageSchema.schema;
 
     expect(required.length).to.eq(1);
-    expect(required[0]).to.eq('myCustomTextInput');
+    expect(required[0]).to.eq(`component${normalizedPage.components[0].id}`);
   });
 
-  context('when component is a text input', () => {
+  describe('selectSchemas', () => {
     let spy;
 
-    beforeEach(() => {
-      spy = sinon.spy(textInput, 'default');
-    });
-
     afterEach(() => {
       spy.restore();
     });
 
-    it('calls the correct function', () => {
-      customStepPage(normalizedPage);
+    [
+      {
+        importedFunction: checkbox,
+        component: {
+          hint: null,
+          id: '172746',
+          label: 'A single checkbox option can be used to confirm agreement',
+          required: true,
+          type: 'digital_form_checkbox',
+          responseOptions: [
+            {
+              id: '172745',
+              label: 'I agree with this statement',
+              description:
+                'I agree to be bound forever and always by this statement.',
+            },
+          ],
+        },
+      },
+      {
+        importedFunction: date,
+        component: {
+          hint:
+            'This date component includes the day as well as the month and the year.',
+          id: '172748',
+          label: 'Date with Day',
+          required: true,
+          type: 'digital_form_date_component',
+          dateFormat: 'month_day_year',
+        },
+      },
+      {
+        importedFunction: radioButton,
+        component: {
+          hint: 'This radio component has two options.',
+          id: '172742',
+          label: 'Test radio component',
+          required: false,
+          type: 'digital_form_radio_button',
+          responseOptions: [
+            {
+              id: '172743',
+              label: 'My custom option',
+              description: 'This option has optional description text.',
+            },
+            {
+              id: '172744',
+              label: 'A second option',
+              description: null,
+            },
+          ],
+        },
+      },
+      {
+        importedFunction: textArea,
+        component: {
+          hint: 'This is optional hint text',
+          id: '172747',
+          label: 'Custom text area',
+          required: false,
+          type: 'digital_form_text_area',
+        },
+      },
+      {
+        importedFunction: textInput,
+        component: {
+          hint: 'This is optional hint text',
+          id: '172748',
+          label: 'Custom text input',
+          required: true,
+          type: 'digital_form_text_input',
+        },
+      },
+    ].forEach(({ importedFunction, component }) => {
+      context(`when component type is ${component.type}`, () => {
+        beforeEach(() => {
+          spy = sinon.spy(importedFunction, 'default');
```

Comment:
TIL meta-programming in Javascript is doable but a huge pain and usually not worth it.